src/material-experimental/form-field/testing/form-field-harness.ts

```diff
@@ -0,0 +1,231 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ComponentHarness,
+  HarnessEnvironment,
+  HarnessPredicate,
+  TestElement
+} from '@angular/cdk/testing';
+import {MatInputHarness} from '@angular/material-experimental/input/testing';
+// TODO(devversion): we cannot import the select harness through the module name
+// because it does not expose an entry-point that re-exports the harness
+import {MatSelectHarness} from '../../mdc-select/harness/select-harness';
+import {FormFieldHarnessFilters} from './form-field-harness-filters';
+
+// TODO(devversion): support datepicker harness once developed (COMP-203).
+// Also support chip list harness.
+/** Possible harnesses of controls which can be bound to a form-field. */
+export type FormFieldControlHarness = MatInputHarness|MatSelectHarness;
```

Comment:
Hmm, people can also make their own custom form field controls. I wonder if we can handle this somehow...

Reply:
Good point. I assume that we can't really detect the harness for custom controls (if there is one), but we can document that for custom form-field controls, the harness needs to be retrieved manually.

Otherwise we could also allow passing in a harness into `getControl(type)` that will return the harness instance for an element that matches the passed harness type. WDYT?

Reply:
Yeah that might be a good idea, we could make the type optional. If you don't pass anything it can just look for input, select, or datepicker. If you do pass one it will look for just the harness you passed. 

If there's any common methods that the `MatFormFieldHarness` would need to call we can make a `MatFormFieldControlHarness` base class and require people to extend that

Reply:
Yes. I like the idea of having a base class for custom form-field control harnesses. I added one that people can implement (even though it is empty for now), but it makes breaking changes less if we ever add stuff there.